### PR TITLE
DL-#7 - Add Dockerfile for php 7.4 from Alpine with Imagemagick insta…

### LINKED ISSUE #7: Create Dockerfile for php
**Description**

Create a Dockerfile which contains the current up-to-date php version and supports to use imagemagick.

**Acceptance criteria**

- PHP 7.4
- Uses the official php-related alpine image
- Pushed to docker hub
- Imagemagick configured

Take into consideration that the base image, aka application image, should not contain any non-production related library, code or any source regarding that consideration. It has to be production ready.

### PR DESCRIPTION
**Pull request by herczogzoltan**
Closes #7

### CODE REVIEW

**Comment by herczogzoltan** [comment]
How to check:

1. Clone the branch and navigate to `php/7.4` directory
3. Build the image locally `docker build --tag docker-library/php-7.4 .`
4. Run it using bash `docker run -ti docker-library/php-7.4 bash`
5. Check if imagemagick installed `apk info imagemagick`
6. Check if imagemagick php library is enabled `php -m | grep imagic`

File: php/7.4/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM php:7.4-fpm-alpine
+
+RUN apk upgrade --update \
+    && apk --update add --no-cache --virtual .phpize-deps $PHPIZE_DEPS imagemagick-dev libtool \
+    && pecl install imagick \
+    && docker-php-ext-configure exif \
+    && docker-php-ext-install -j$(nproc) mysqli \
```
Comment (saxumVermes) [inline php/7.4/Dockerfile]: docker-php-ext-install can recieve multiple extensions as arguments.

File: php/7.4/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    && docker-php-ext-install -j$(nproc) exif \
+    && docker-php-ext-install -j$(nproc) pdo pdo_mysql \
```
Comment (saxumVermes) [inline php/7.4/Dockerfile]: Like you did it here

**Review by saxumVermes — CHANGES_REQUESTED** [review]

File: php/7.4/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
+FROM php:7.4-fpm-alpine
+
+RUN apk upgrade --update \
+    && apk --update add --no-cache --virtual .phpize-deps $PHPIZE_DEPS imagemagick-dev libtool \
```
Comment (saxumVermes) [inline php/7.4/Dockerfile:4]: Could you put a bash into the container in a separate apk add command while also merging imagemagick from line 12 into it?

File: php/7.4/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    && pecl install imagick \
+    && docker-php-ext-configure exif \
+    && docker-php-ext-install -j$(nproc) mysqli \
+    && docker-php-ext-install -j$(nproc) exif \
+    && docker-php-ext-install -j$(nproc) pdo pdo_mysql \
+    && docker-php-ext-enable imagick \
+    && docker-php-ext-enable exif \
+    && apk add --no-cache --virtual .imagick-runtime-deps imagemagick \
```
Comment (saxumVermes) [inline php/7.4/Dockerfile]: Also I don't see the point of using virtual package here or is there a reason I'm not aware of?

**Review by saxumVermes — CHANGES_REQUESTED** [review]

File: php/7.4/Dockerfile (hunk 3 above)
Comment (saxumVermes) [reply php/7.4/Dockerfile:4]: Please take a look if this works.
```suggestion
    && apk --update add --no-cache --virtual .phpize-deps $PHPIZE_DEPS imagemagick-dev libtool \
    && apk --update add --no-cache imagemagick bash \
```

Comment (herczogzoltan) [reply php/7.4/Dockerfile:4]: Also I assume line 12 will change to
```
&& apk add --no-cache --virtual .imagick-runtime-deps \
```

Am I assumed it correctly?

Comment (saxumVermes) [reply php/7.4/Dockerfile:4]: It can be deleted

Comment (herczogzoltan) [reply php/7.4/Dockerfile:4]: Alright, I'll check it

Comment (herczogzoltan) [reply php/7.4/Dockerfile:4]: Great! It worked! I'll push my modifications for a re-check.

File: php/7.4/Dockerfile (hunk 4 above)
Comment (herczogzoltan) [reply php/7.4/Dockerfile]: It turned out that from the aspect of `imagemagick` it's not relevant to use virtual package, but it had a reason for `imagemagick-runtime-deps`, the new commit provides a new way of handling dependencies and the cleanup

**Pull request merged**